### PR TITLE
LPS-170550 [EVP] Creating Reports Page Filtering Functionality

### PR DESCRIPTION
**Pull request by juliocezar96**
[LPS-170550](https://issues.liferay.com/browse/LPS-170550)

### CODE REVIEW

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:relevant

**Comment by juliocezar96** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	idOrganizationInitial: number
+) {
+	const response = await fetcher(
+		`${resource}/?filter=idNumber ge ${idOrganizationInitial}`
+	);
+
+	return response;
+}
+
+export async function filteredOrganizationsERC(organization: any) {
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts:34]: Could you put the type in the organizations parameter and change the function name?
```suggestion
export async function filteredOrganizationsERC(organizations: ???) {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+export async function getOrganizations(organizationName: string) {
+	const response = await fetcher(
+		`${resource}/?filter=contains(organizationName,'${organizationName}')`
+	);
+
+	return response;
+}
+
+export async function getIdGreaterOrEqualOrganization(
+	idOrganizationInitial: number
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts:25]:
```suggestion
	organizationId: number
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+export async function getOrganizations(organizationName: string) {
+	const response = await fetcher(
+		`${resource}/?filter=contains(organizationName,'${organizationName}')`
+	);
+
+	return response;
+}
+
+export async function getIdGreaterOrEqualOrganization(
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts:24]:
```suggestion
export async function getGreaterOrEqualOrganizationId(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	const externalReferenceCodes: string[] = [];
+
+	organization.items.forEach((organization: OrganizationFilterType) => {
+		externalReferenceCodes.push(organization.externalReferenceCode);
+	});
+
+	return externalReferenceCodes;
+}
+
+export async function getIdLessOrEqualOrganization(
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts:44]:
```suggestion
export async function getLessOrEqualOrganizationId(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	idOrganizationFinal: number
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts:45]:
```suggestion
	organizationId: number
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	const response = await fetcher(
+		`${resource}/?filter=idNumber le ${idOrganizationFinal}`
+	);
+
+	return response;
+}
+
+export async function getOrganizationBetweenIds(
+	idOrganizationInitial: number,
+	idOrganizationFinal: number
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts:56]:
```suggestion
	initialOrganizationId: number,
	finalOrganizationId: number
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+) {
+	const response = await fetcher(
+		`${resource}/?filter=idNumber ge ${idOrganizationInitial} and idNumber le ${idOrganizationFinal}`
+	);
+
+	return response;
+}
+
+export async function getOrganizationName(organizationName: string) {
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/organization.ts:65]: How is this function different from the first function in this file?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts [hunk 8]
```diff
@@ -10,12 +9,180 @@
...
+	getOrganizationName,
+} from './organization';
 
 const resource = 'o/c/evprequests';
 
-export async function getRequestsByFilter() {
-	const response = await fetcher(`${resource}`);
+const nestedFields = '&nestedFields=r_organization_c_evpOrganization';
+
+const formatArrayUrl = (key: string, value: String[]) => {
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts:28]:
```suggestion
const formatArrayUrl = (key: string, values: String[]) => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts [hunk 9]
```diff
@@ -10,12 +9,180 @@
...
+	let operator = `${key} in ({values})`;
+
+	operator = operator
+		.replace('{values}', value.map((value) => `'${value}'`).join(','))
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts:32]: Could you put the type to value parameter?
```suggestion
		.replace('{values}', value.map((value: ???) => `'${value}'`).join(','))
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts [hunk 10]
```diff
@@ -10,12 +9,180 @@
...
+		filter.push(formatUrl);
+	}
+
+	if (data.liferayBranch.length !== 0) {
+		formatUrl = formatArrayUrl(
+			FIELDSREPORT.LIFERAYBRANCH,
+			data.liferayBranch
+		);
+		filter.push(formatUrl);
+	}
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts:69]: There is the same behavior between these conditions. Please turn them into a generic function.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts [hunk 11]
```diff
@@ -10,12 +9,180 @@
...
+		);
+
+		if (externalReferenceCodeOrganization.length !== 0) {
+			formatUrl = formatArrayUrl(
+				'r_organization_c_evpOrganizationERC',
+				externalReferenceCodeOrganization
+			);
+			filter.push(formatUrl);
+		}
+	}
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts:131]: Please turn them into a generic function as well.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts [hunk 12]
```diff
@@ -10,12 +9,180 @@
...
+		}
+		else {
+			formattedTimeFinalRequestDate = finalFormatRequestTime(
+				ISOFormattedFinalRequestDate
+			);
+
+			filter.push(
+				`dateCreated ge ${ISOFormattedInitialRequestDate} and dateCreated le ${formattedTimeFinalRequestDate}`
+			);
+		}
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts:158]: The `if` and `else` conditions have the same behavior here. Please remove it.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx [hunk 13]
```diff
@@ -66,50 +64,49 @@ const GenerateReport = () => {
...
 	const onClickStatus = (event: any) => {
-		const value = Number(event.target.value);
-
+		const value = event.target.value;
 		const statusesFiltered = statusesWatch?.includes(value)
 			? statusesWatch.filter((statusId) => statusId !== value)
 			: [...statusesWatch, value];
 
-		setValue('statusRequest', statusesFiltered);
+		setValue('requestStatus', statusesFiltered);
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx:82]: Please turn these functions into a generic function.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx [hunk 14]
```diff
@@ -158,38 +146,31 @@ const GenerateReport = () => {
...
-						/>
-					</div>
 				</div>
 
 				<div className="row">
 					<div className="col">
 						<label>Statuses</label>
 
-						{statuses.map((status: any, index: string) => (
+						{statuses.map((status: Statuses, index: string) => (
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx:160]: What kind is Statuses? Please rename this type to Statustype

Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx:160]: Why is index being typed as a string?
```suggestion
						{statuses.map((status: Statuses, index: string ???) => (
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx [hunk 15]
```diff
@@ -198,21 +179,25 @@ const GenerateReport = () => {
...
-									id="branchRequest"
-									label={branch.name}
-									name="branchRequest"
-									onChange={onClickBranches}
-									value={branch.id}
-								/>
-							</div>
-						))}
+						{branches.map(
+							(branch: LiferayBranch, index: string) => (
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx:183]: Why is index being typed as a string?
```suggestion
							(branch: LiferayBranch, index: string ???) => (
```

Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/routes/reports/generateReport/index.tsx:183]: What kind is LiferayBranch? Please rename this type to LiferayBranchType
```suggestion
							(branch: LiferayBranchType, index: string) => (
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/types/index.ts [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+	organizationStatus: string[];
+	smallDescription: string;
+	state: string;
+	status: string[];
+	street: string;
+	taxIdentificationNumber: string;
+	zip: string;
+}
+
+export type Statuses = {
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/types/index.ts:41]: The name of this type doesn't make sense to me. Can you explain me?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/types/index.ts [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+	dateCreated: string;
+	dateModified: string;
+	id: number;
+	key: string;
+	name: string;
+};
+
+export type LiferayBranch = {
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/types/index.ts:49]: The name of this type doesn't make sense to me. Can you explain me?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/organization.json [hunk 18]
```diff
@@ -177,6 +178,24 @@
...
 		},
+		{
+			"DBType": "Integer",
+			"businessType": "Integer",
+			"defaultValue": "",
+			"indexed": true,
+			"indexedAsKeyword": false,
+			"indexedLanguageId": "",
+			"label": {
+				"en_US": "IDNumber"
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/organization.json:189]: Please, changet it.
```suggestion
				"en_US": "ID Number"
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/organization.object-actions.json [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+		"objectActionExecutorKey": "update-object-entry",
+		"objectActionTriggerKey": "onAfterAdd",
+		"objectDefinitionId": "[$OBJECT_DEFINITION_ID:EVPOrganization$]",
+		"parameters": {
+			"objectDefinitionExternalReferenceCode": "OBJECT-EVP-ORGANIZATION",
+			"predefinedValues": [
+				{
+					"inputAsValue": false,
+					"label": {
+						"en_US": "IDNumber"
```
Comment (silasbsi) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/organization.object-actions.json:14]: Please, change it.
```suggestion
						"en_US": "ID Number"
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

**Comment by silasbsi** [comment]
Please rename the commits. They are the same.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/components/Form/DatePicker/index.tsx [hunk 20]
```diff
@@ -22,19 +21,24 @@ type DatePickerTypes = {
 	name: string;
 	register?: any;
 	required?: boolean;
+	setValue?: any;
```
[the remark this answers is not in the document]
Comment (JoMaAlves) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/components/Form/DatePicker/index.tsx:24]: (item: string) => void;

 const fuu = (item: string) => {
    setValue(item);
}

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts (hunk 10 above)
Comment (JoMaAlves) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/extra/remote-app/src/common/services/request.ts:69]: const fuu = (x, y) => {
   formatUrl = formatArrayUrl(
			x,
			y
		);
    filter.push(formatUrl);
}

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:relevant

**Review by silasbsi — APPROVED** [review]
LGTM :heavy_check_mark:

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**